docs/yaml/functions/response_file.yaml

```diff
@@ -0,0 +1,67 @@
+name: response_file
+returns: rsp_file
+description: |
+  Write argument list to a response file, if the resulting length of the command
+  line is longer than a maximum value. A response file is a mechanism allowing a
+  program to receive more arguments than what would be allowed by the command
+  line. See [msvc compiler
+  documentation](https://learn.microsoft.com/en-us/cpp/build/reference/at-specify-a-compiler-response-file?view=msvc-170)
+  as an example.
+
+  This function is designed to be used with [[custom_target]], [[run_target]],
+  [[run_command]] or [[configure_file]].
+
+varargs:
+  name: arguments
+  type: str | file
+  description: The argument list.
```

Comment:
I suspect people will want to use outputs of `custom_target`s et al in this too. That would require for us to set the dependencies to said targets as well so they are up to date. But that can probably be added later.

Reply:
It accepts BuildTarget, CustomTarget, and CustomTargetIndex as well. I just forgot to update the documentation. Will do.

Reply:
What about generator outputs?

Reply:
I feel like we need to refactor completely the command args handling in Meson before extending it with response_file(). I had 2 old PRs attempting at cleaning the current mess: https://github.com/mesonbuild/meson/pull/11061 and https://github.com/mesonbuild/meson/pull/8342.

I'll see if I can restart that work tomorrow.

Reply:
> I had 2 old PRs attempting at cleaning the current mess: #11061 and #8342 and #6624.

There are three kind of people in the world: those who are able to count and those who don't...

I understand your concern, and I agree with you (this is one reason why my `response_file` function does not handle `test` for now...) However, I also really need this feature for my project. I wonder whether we should wait for perfection, or start with something incomplete, but working for most usecases... Maybe it could be documented as *unstable*, just like modules can be...